src/source/lexstring.d

```diff
@@ -551,11 +548,23 @@ unittest {
 
 	// Check octal escape sequences.
 	checkLexString(`"\0\1\11\44\77\111\377"`, "\0\x01\x09\x24\x3f\x49\xff");
-	checkLexString(`"\1111\378"`, "\x491\x1f8");
+	checkLexString(`"\0\00\000"`, "\0\00\000");
```

Comment:
We should test minimum and maximum acceptable number, prefixed with one or two zero, and if appropriate the next number in line, so we check for boundary condition.

Also what happens if we do things like '\0377' ? Before, we didn't really have that edge case as the octal digits would always be too large if they were more than 3 digits, but not anymore.